openeogeotrellis/backend.py

```diff
@@ -2022,8 +2031,20 @@ def as_arg_element(dependency: dict) -> dict:
                 yunikorn_queue=get_backend_config().yunikorn_queue,
                 yunikorn_scheduling_timeout=get_backend_config().yunikorn_scheduling_timeout.rstrip(),
                 try_swift_streaming=os.environ.get("TRY_SWIFT_STREAMING"),
+                provide_s3_profiles_and_tokens=get_backend_config().provide_s3_profiles_and_tokens,
+                batch_job_cfg_secret_name=batch_job_cfg_secret_name
             )
 
+            if get_backend_config().provide_s3_profiles_and_tokens:
+                # For now we cannot access the subject of the initial access token but generally it is the user_id
+                s3_profiles_cfg_batch_secret = k8s_render_manifest_template(
+                    "batch_job_cfg_secret.yaml.j2",
+                    secret_name=batch_job_cfg_secret_name,
+                    job_id=job_id,
+                    token=IDP_TOKEN_ISSUER.get_job_token(sub_id=user_id, user_id=user_id, job_id=job_id),
+                    profile_file_content=S3Config.from_backend_config(job_id, "/opt/job_config/token")
```

Comment:
This file is provided by the same secret. Kubernetes will mount his secret and the token file will appear. I did however push another commit d919872b286af2e56651c85453bf3b88217ce5b2  because the `/opt/job_config` is actually used in multiple locations so better extracted in a config var.